include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -472,46 +472,62 @@ template <typename _ContainerOrIterable>
 using __value_t = typename __internal::__memobj_traits<_ContainerOrIterable>::value_type;
 
 template <typename _T>
-struct __usm_host_or_buffer_accessor
+struct __usm_or_buffer_accessor
 {
   private:
     using __accessor_t = sycl::accessor<_T, 1, sycl::access::mode::read_write, __dpl_sycl::__target_device,
                                         sycl::access::placeholder::false_t>;
     __accessor_t __acc;
     _T* __ptr = nullptr;
     bool __usm = false;
+    size_t __offset;
```

Comment:
If we write here ` = 0`

Reply:
I think you are pointing out an interesting choice between default assignment vs initializer list for this class with your comments. I agree it should be consistent, however I would instead remove the default assignment values and only use the initializer list. As all of the values change based upon the specific constructor called, if they did not change based on the constructor then I would use default assignment.

So I am proposing removing any default assignment values and only using initializer list. Would this address your comments, or due you still prefer the default assignment?

Reply:
Probably you will have the same initializations in different constructors in this case, as `__offset(0)` now.
I would like to propose a little bit different variant: add ` = 0` here and in constructors initialize only these fields which really required the different values, not as on declaration.

Reply:
Iv followed this approach